File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
    
    function dfs(count, num){
        if(count === numbers.length) {
            if(num === target) answer += 1;
            return;
        }
        
        dfs(count + 1, num + numbers[count])
        dfs(count + 1, num - numbers[count])
    }
    
    dfs(0, 0)
    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    
    def dfs(count, sum):
        if count==len(numbers):
            if sum==target:
                nonlocal answer
                answer+=1
            return
    
        dfs(count+1, sum+numbers[count])
        dfs(count+1, sum-numbers[count])
    
    dfs(0, 0)
    return answer

